include/ck_tile/core/arch/arch.hpp

```diff
@@ -82,14 +82,12 @@ CK_TILE_DEVICE void block_sync_lds_direct_load()
     " ::);
 }
 
-CK_TILE_DEVICE void s_nop()
+CK_TILE_DEVICE void s_nop(index_t cnt = 0)
 {
 #if 1
-    asm volatile("\
-    s_nop 0 \n \
-    " ::);
+    asm volatile("s_nop %0" : : "n"(cnt) :);
 #else
-    __builtin_amdgcn_sched_barrier(0);
+    __builtin_amdgcn_sched_barrier(cnt);
```

Comment:
Will this have the same effect? https://llvm.org/docs/AMDGPUUsage.html#llvm-ir-intrinsics

Reply:
actually there is no intrinsic for s_nop, so usually will use inline asm version. If we want to use `__builtin_amdgcn_sched_barrier` usually just directly use this. So here you can ignore the #else